Sources/XCTest/XCTestObservation.swift

```diff
@@ -0,0 +1,58 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//
+//  XCTestObservation.swift
+//  WRITEME
+//
+
+public protocol XCTestObservation: class {
+    func testCaseWillStart(testCase: XCTestCase)
+    func testCase(testCase: XCTestCase, didFailWithDescription description: String, inFile filePath: String?, atLine lineNumber: UInt)
+    func testCaseDidFinish(testCase: XCTestCase)
+}
+
+
+
+public class XCTestObservationCenter {
+    private static var center = XCTestObservationCenter()
+    private var observers = [XCTestObservation]()
```

Comment:
Unfortunately not, because `XCTestObservation` doesn't conform to `Hashable`. I briefly considered `NSMutableSet` but I'm not convinced that it's worth losing the strong typing.